File: auctions-service/src/main/java/com/trato/auction/common/controller/HealthController.java
package com.trato.auction.common.controller;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import java.util.Map;

@RestController
public class HealthController {
    @GetMapping("/health")
    public Map<String, Object> health() {
        return Map.of("success", true, "message", "Auctions service is healthy");
    }
}

File: microservices/auth-service/src/health.controller.ts
import { Controller, Get } from '@nestjs/common';

@Controller()
export class HealthController {
  @Get('health')
  health() {
    return {
      success: true,
      data: {
        status: 'ok',
        service: 'auth-service',
      },
    };
  }
}
